### PR TITLE
fix: find highlight disappears on clicking the cell

### LINKED ISSUE #14222: Notebook find highlight disappears when clicked on any markdown cell.
Issue Type: <b>Bug</b>

Open a notebook document, launch find ctrl-f and search for a term. And click on any cell. Find highloght disappears even though then find widget is active.

Azure Data Studio version: azuredatastudio 1.25.2 (4d4917d328ab7a7526817a0f0ef03237a79aa54d, 2021-01-20T22:47:41.882Z)
OS version: Darwin x64 20.1.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 4|
|Memory (System)|16.00GB (2.99GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

### PR DESCRIPTION
**Pull request by MaddyDev**
This PR fixes #14222

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 1]
```diff
@@ -93,6 +93,7 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 	public readonly onDidClickLink = this._onDidClickLink.event;
 	public previewFeaturesEnabled: boolean = false;
 	public doubleClickEditEnabled: boolean;
+	private highlightRange: NotebookRange;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts:96]: Convention is that private vars start with `_`

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 2]
```diff
@@ -344,15 +346,18 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 
 	public deltaDecorations(newDecorationRange: NotebookRange, oldDecorationRange: NotebookRange): void {
 		if (oldDecorationRange) {
+			this.highlightRange = oldDecorationRange === this.highlightRange ? null : this.highlightRange;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts:349]: Use undefined, not null

**Review by Charles-Gagnon — APPROVED** [review]
Wouldn't be a bad idea to add tests for this

**Review by halerankin — COMMENTED** [review]
I noticed on Mac OS X, with a given cell editor open, that the **scroll to** position of the current highlighted hit (in orange) is just out of view. See image below: 
image

I will add, that when the editor is NOT open, the highlighted hit is not obstructed by the UI in any way. See image below:
image

**Comment by chlafreniere** [comment]
@MaddyDev, can we also look into not calling through to the markdown renderer code on cell selection?

**Review by corivera — APPROVED** [review]

**Comment by MaddyDev** [comment]
> I noticed on Mac OS X, with a given cell editor open, that the **scroll to** position of the current highlighted hit (in orange) is just out of view. See image below:
> image
> 
> I will add, that when the editor is NOT open, the highlighted hit is not obstructed by the UI in any way. See image below:
> image

Can you open an issue for this and assign it to me. Thanks!

**Pull request merged**